addons/account/models/account_bank_statement.py

```diff
@@ -623,6 +623,14 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
             total -= aml_rec.debit - aml_rec.credit
             aml_rec.with_context(check_move_validity=False).write({'statement_line_id': self.id})
             counterpart_moves = (counterpart_moves | aml_rec.move_id)
+            if aml_rec.journal_id.post_at_bank_rec and aml_rec.payment_id and aml_rec.move_id.state == 'draft':
+                # In case the journal is set to only post payments when performing bank
+                # reconciliation, we modify its date and post it.
+                aml_rec.move_id.date = self.date
+                aml_rec.payment_id.payment_date = self.date
+                aml_rec.move_id.post()
+                # We check the paid status of the invoices reconciled with this payment
+                aml_rec.payment_id.reconciled_invoice_ids.filtered(lambda x: x.state == 'in_payment').action_invoice_paid()
```

Comment:
it's plain wrong to mark invoices in 'paid' status: it depends on the amount_residual to keep them in 'open' status or not